pkg/maintenance/helm/helmpatcher_test.go

```diff
@@ -458,3 +458,162 @@ func getResult(n, s, ct string) Result {
 		ContentType: ct,
 	}
 }
+
+func Test_minorVersionUpgrades(t *testing.T) {
+	defaultV := "7.0"
+	tests := []struct {
+		name              string
+		instanceNamespace string
+		release           *v1beta1.Release
+		results           []Result
+		expectedVer       string
+		expectedNew       bool
+		expectedErr       string
+	}{
+		{
+			name:              "WhenNewVersionFromReleases_ThenGetNewVersion",
+			instanceNamespace: "test-namespace",
+			release: &v1beta1.Release{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test-release",
+				},
+				Spec: v1beta1.ReleaseSpec{
+					ForProvider: v1beta1.ReleaseParameters{
+						ValuesSpec: v1beta1.ValuesSpec{
+							Values: runtime.RawExtension{
+								Raw: []byte("{\"image\":{\"tag\":\"7.0\"}}"),
+								Object: &runtime.Unknown{
+									Raw: []byte("{\"image\":{\"tag\":\"7.0\"}}"),
+								},
+							},
+						},
+					},
+				},
+			},
+			results: []Result{
+				getResult("6.5.4", "active", "image"),
+				getResult("6.6.4", "inactive", "image"),
+				getResult("6.8", "active", "image"),
+				getResult("7.0", "active", "image"),
+				getResult("7.0.1", "active", "registry"),
+				getResult("7.0.4", "active", "image"),
+				getResult("7.1.4", "active", "image"),
+				getResult("7.2.4", "inactive", "image"),
+				getResult("7.1.12-alpha", "active", "image"),
+				getResult("7.3.11", "active", "image"),
+				getResult("7.3.12", "active", "image"),
+				getResult("7.3.13", "active", "image"),
+				getResult("7.3.14", "active", "image"),
+				getResult("7.0.13", "active", "image"),
+				getResult("7.0.13-debian", "active", "image"),
+				getResult("7.0.14-alpha", "active", "image"),
+				getResult("7.0.14%alpa", "active", "image"),
+			},
+			expectedVer: semver.MustParse("7.3.14").String(),
+		},
+		{
+			name:              "WhenNoNewVersion_ThenReleaseNoNewVersion",
+			instanceNamespace: "test-namespace",
+			release: &v1beta1.Release{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test-release",
+				},
+				Spec: v1beta1.ReleaseSpec{
+					ForProvider: v1beta1.ReleaseParameters{
+						ValuesSpec: v1beta1.ValuesSpec{
+							Values: runtime.RawExtension{
+								Raw: []byte("{\"image\":{\"tag\":\"7.0\"}}"),
+								Object: &runtime.Unknown{
+									Raw: []byte("{\"image\":{\"tag\":\"7.0\"}}"),
+								},
+							},
+						},
+					},
+				},
+			},
+			results: []Result{
+				getResult("6.5.4", "active", "image"),
+				getResult("6.6.4", "inactive", "image"),
+				getResult("6.8", "active", "image"),
+				getResult("7.0", "active", "image"),
+				getResult("7.1.4", "active", "image"),
+				getResult("7.2.4", "inactive", "image"),
+				getResult("7.1.12-alpha", "active", "image"),
+				getResult("7.3.11", "active", "image"),
+				getResult("7.0.14-alpha", "active", "image"),
+				getResult("7.0.14%alpa", "active", "image"),
+			},
+			expectedVer: "7.3.11",
```

Comment:
hmm I think you duplicated here the test. it should stay the same old version?

Reply:
You're right, it should actually return the same string as in the values.

Probably a copy paste error from when the first strategy was implemented, I've now adjusted the test so that there's actually really no newer version in the list.